src/planning.py

```diff
@@ -662,6 +717,110 @@ def _get_successor_with_correct_option(
     return action_seq
 
 
+def _sesame_plan_with_fast_downward(
+        task: Task, option_model: _OptionModelBase, nsrts: Set[NSRT],
+        initial_predicates: Set[Predicate], types: Set[Type], timeout: float,
+        seed: int, max_horizon: int,
+        optimal: bool) -> Tuple[List[_Option], Metrics]:
+    """A version of SeSamE that runs the Fast Downward planner to produce a
+    single skeleton, then calls run_low_level_search() to turn it into a plan.
+
+    Usage: Build and compile the Fast Downward planner, then set the environment
+    variable FD_EXEC_PATH to point to the `downward` directory. For example:
+    1) git clone https://github.com/ronuchit/downward.git
+    2) cd downward && ./build.py
+    3) export FD_EXEC_PATH="<your path here>/downward"
+    """
+    # Note: the types that would be extracted from the NSRTs here may not
+    # include all the environment's types, so it's better to use the
+    # types that are passed in as an argument instead.
+    nsrt_preds, _ = utils.extract_preds_and_types(nsrts)
```

Comment:
why do you need to do this at all? can't we use the predicates that are passed in?

Reply:
i know we had a good reason for this before, but maybe it's out of date. let's just try removing it and see what happens 🤷

Reply:
i bet the discovered failure thing will break